var/spack/repos/builtin/packages/py-ipywidgets/package.py

```diff
@@ -12,6 +12,7 @@ class PyIpywidgets(PythonPackage):
     homepage = "https://github.com/ipython/ipywidgets"
     url      = "https://github.com/ipython/ipywidgets/archive/5.2.2.tar.gz"
 
+    version('7.4.2', sha256='f156165e8a855ed862fdf48e72700bdcd6956d089a2018c5b36d358255d45b2b')
     version('5.2.2', sha256='d61ab8bb12b90981a3a6010429816d70eaa041e622043207bcb74239b664d4f3')
 
     depends_on('python@2.7:2.8,3.3:')
```

Comment:
```suggestion
    depends_on('python@2.7:2.8,3.3:', type=('build', 'run'))
```

Reply:
There's also some changes to the dependencies. You'll have to see which version added these:
```python
install_requires = setuptools_args['install_requires'] = [                      
    'ipykernel>=4.5.1',                                                         
    'traitlets>=4.3.1',                                                         
    # Requiring nbformat to specify bugfix version which is not required by     
    # notebook.                                                                 
    'nbformat>=4.2.0',                                                          
    # TODO: Dynamically add this dependency                                     
    # only if notebook 4.x is installed in this                                 
    # interpreter, to allow ipywidgets to be                                    
    # installed on bare kernels.                                                
    'widgetsnbextension~=3.4.0',                                                
]                                                                               
                                                                                
extras_require = setuptools_args['extras_require'] = {                          
    ':python_version<"3.3"' : ['ipython>=4.0.0,<6.0.0'],                        
    ':python_version>="3.3"': ['ipython>=4.0.0'],                               
    'test:python_version=="2.7"': ['mock'],                                     
    'test': ['nose'],                                                           
}                     
```

Reply:
Don't forget to add these dependencies.